presto-jdbc/src/main/java/io/prestosql/jdbc/PrestoPreparedStatement.java

```diff
@@ -453,7 +459,35 @@ else if (x instanceof Timestamp) {
     public void addBatch()
             throws SQLException
     {
-        throw new NotImplementedException("PreparedStatement", "addBatch");
+        checkOpen();
+        batchValues.add(toValues(parameters));
+        isBatch = true;
+    }
+
+    @Override
+    public void clearBatch()
+            throws SQLException
+    {
+        checkOpen();
+        batchValues.clear();
+        isBatch = false;
+    }
+
+    @Override
+    public int[] executeBatch()
+            throws SQLException
+    {
+        try {
+            int[] batchUpdateCounts = new int[batchValues.size()];
+            for (int i = 0; i < batchValues.size(); i++) {
+                super.execute(getExecuteSql(statementName, batchValues.get(i)));
```

Comment:
is it possible execute SQLs in batch as well instead of one by one ?
ex: in case of INSERT, one by one insertion cause small files (hive connector)

Reply:
That would be nice to have, but i'd rather see this as a follow up to this change.

Reply:
> That would be nice to have, but i'd rather see this as a follow up to this change.

Is that implemented already?

Reply:
Not yet.

Reply:
@ebyhr  does trino server support batching?